lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
Okay, so we need an object to hold the data before the worker constructs the entity. Then we could make a specific `ContactRetraction` object that encapsulates all of the specifics of contact retraction and avoid branching in entity dispatching process. It's pretty much the same thing as what you do, but it uses inheritance instead of branching which is a bit cleaner IMO.

Example: https://gist.github.com/cmrd-senya/90a4bc2aaf7370b24caa91e5487fa763

What do you think?